### PR TITLE
Adds webhook for KIT

### PR DESCRIPTION
**Pull request by prateekgogia**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: operator/Makefile [hunk 1]
```diff
@@ -46,9 +46,13 @@ licenses: ## Verifies dependency licenses and requires GITHUB_TOKEN to be set
 
 apply: ## Apply the controller into your ~/.kube/config cluster
 	KO_DOCKER_REPO=$(CONTAINER_IMAGE_REGISTRY)/kit ko apply --bare -f config/
```
Comment (ellistarn) [inline operator/Makefile:48]: Do you need this line, too?

File: operator/Makefile [hunk 2]
```diff
@@ -46,9 +46,13 @@ licenses: ## Verifies dependency licenses and requires GITHUB_TOKEN to be set
...
+	KO_DOCKER_REPO=$(CONTAINER_IMAGE_REGISTRY)/kit ko apply --bare -f config/controller
+	KO_DOCKER_REPO=$(CONTAINER_IMAGE_REGISTRY)/kit ko apply --bare -f config/webhook
     
 delete: ## Delete the controller from your ~/.kube/config cluster
 	kubectl delete -f config
```
Comment (ellistarn) [inline operator/Makefile:53]: likewise

File: operator/cmd/webhook/main.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
+/*
```
Comment (ellistarn) [inline operator/cmd/webhook/main.go:1]: Checkout the latest in Karpenter in this file. There are some improvements.

File: operator/config/controller/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        resources:
+          requests:
+            cpu: 100m
+            memory: 20Mi
+      # https://github.com/aws/amazon-eks-pod-identity-webhook/issues/8#issuecomment-636888074
+      securityContext:
+        fsGroup: 1000
+      nodeSelector:
+        kubernetes.io/hostname: ip-192-168-12-204.us-west-2.compute.internal
+      tolerations:
```
Comment (ellistarn) [inline operator/config/controller/deployment.yaml:31]: Why this?

**Review by ellistarn — APPROVED** [review]

File: operator/Makefile (hunk 1 above)
Comment (prateekgogia) [reply operator/Makefile:48]: I wanted to separate the control plane CRD yaml spec from the controller. So config/ contains this spec, controller and webhook are subdirectories that contain their specific yaml files.

File: operator/config/controller/deployment.yaml (hunk 4 above)
Comment (prateekgogia) [reply operator/config/controller/deployment.yaml:31]: Nice catch! I forgot to remove this I was using this for local testing. I will open another PR to remove this.

**Pull request merged**